File: Concentus.Celt/Structs/AnalysisInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concentus.Celt.Structs
{
    public class AnalysisInfo
    {
        public int valid = 0;
        public float tonality = 0;
        public float tonality_slope = 0;
        public float noisiness = 0;
        public float activity = 0;
        public float music_prob = 0;
        public int bandwidth = 0;

        public AnalysisInfo()
        {
        }

        public void Assign(AnalysisInfo other)
        {
            this.valid = other.valid;
            this.tonality = other.tonality;
            this.tonality_slope = other.tonality_slope;
            this.noisiness = other.noisiness;
            this.activity = other.activity;
            this.music_prob = other.music_prob;
            this.bandwidth = other.bandwidth;
        }

        public void Reset()
        {
            valid = 0;
            tonality = 0;
            tonality_slope = 0;
            noisiness = 0;
            activity = 0;
            music_prob = 0;
            bandwidth = 0;
        }
    }
}

File: Go/celt/analysisInfo.go
package celt

type AnalysisInfo struct {
	Enabled        bool
	Valid          int
	Tonality       float32
	Tonality_slope float32
	Noisiness      float32
	Activity       float32
	Music_prob     float32
	Bandwidth      int
}

func (a *AnalysisInfo) Assign(other *AnalysisInfo) {
	a.Valid = other.Valid
	a.Tonality = other.Tonality
	a.Tonality_slope = other.Tonality_slope
	a.Noisiness = other.Noisiness
	a.Activity = other.Activity
	a.Music_prob = other.Music_prob
	a.Bandwidth = other.Bandwidth
}

func (a *AnalysisInfo) Reset() {
	a.Valid = 0
	a.Tonality = 0
	a.Tonality_slope = 0
	a.Noisiness = 0
	a.Activity = 0
	a.Music_prob = 0
	a.Bandwidth = 0
}
